cartographer_ros/cartographer_ros/node.cc

```diff
@@ -67,10 +67,15 @@ ::rclcpp::SubscriptionBase::SharedPtr SubscribeWithHandler(
                           const typename MessageType::ConstSharedPtr),
     const int trajectory_id, const std::string& topic,
     ::rclcpp::Node::SharedPtr node_handle, Cartographer* const node) {
+  auto last_message_time = std::make_shared<rclcpp::Time>(node_handle->get_clock()->now());
```

Comment:
I'm concerned that we are (potentially) comparing the time across two different time sources.  If the sensor message is being published from another computer, than we are comparing the timestamp on this computer (`get_clock()->now()`) with a timestamp made on another computer (`msg->header.stamp`).

Instead of calling `get_clock()->now()`, what if we just stored `msg->header.stamp` from the previous message, and then compare if it is newer than that?  We'd also have to deal with the initial case where we don't have a time, but we can start it at 0 or something like that.

(this doesn't fix the case where we have two different computers publishing to the same topic, but I'm guessing that is going to lead to havoc in other ways, so I'm ignoring it here)

Reply:
Sounds reasonable.

Reply:
See https://github.com/ros2/cartographer_ros/pull/46/commits/2baa23bd931180fcd0bcc095573ccab116099de9.